TestProject.BL.Test/Services/PostServiceTests.cs

```diff
@@ -0,0 +1,134 @@
+﻿using Moq;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using TestProject.BL.Mappers;
+using TestProject.BL.Models;
+using TestProject.BL.Services;
+using TestProject.DAL.Models;
+using TestProject.DAL.Repositories;
+using Xunit;
+using FluentAssertions;
+using TestProject.BL.Exceptions;
+
+namespace TestProject.BL.Test.Services
+{
+    public class PostServiceTests
+    {
+        private PostService _postService;
+        private Mock<IRepository<Post>> _mockPostRepository;
+        private Mock<IRepository<User>> _mockUserRepository;
+        private Mock<IMapper<EditPostModel, Post>> _mockEditPostMapper;
+        private Mock<IMapper<PostModel, Post>> _mockPostMapper;
+
+        public PostServiceTests()
+        {
+            _mockPostRepository = new Mock<IRepository<Post>>();
+            _mockUserRepository = new Mock<IRepository<User>>();
+            _mockEditPostMapper = new Mock<IMapper<EditPostModel, Post>>();
+            _mockPostMapper = new Mock<IMapper<PostModel, Post>>();
+            _postService = new PostService(_mockPostRepository.Object, _mockUserRepository.Object, _mockEditPostMapper.Object, _mockPostMapper.Object);
+        }
+
+        [Fact]
+        public async Task Should_create_new_post()
+        {
+            var post = new Post();
+            var users = new List<User>() { new User() };
+            _mockEditPostMapper.Setup(mapper => mapper.ToDalModel(It.IsAny<EditPostModel>())).Returns(post);
+            _mockUserRepository.Setup(repo => repo.Get(It.IsAny<Func<User, bool>>())).Returns(users);
+
+            await _postService.Create(new EditPostModel(), "");
+
+            _mockPostRepository.Verify(repo => repo.Create(post), Times.Exactly(1));
+        }
+
+        [Theory]
+        [MemberData(nameof(GetAllData))]
+        public async Task Should_get_posts(int? userId, List<PostModel> expected, Post post, PostModel postModel)
+        {
+            _mockPostRepository.Setup(repo => repo.Get(It.IsAny<Func<Post, bool>>())).Returns(new List<Post>() { post });
+            _mockPostRepository.Setup(repo => repo.Get()).ReturnsAsync(new List<Post>() { post, post });
+            _mockPostMapper.Setup(mapper => mapper.ToBlModel(post)).Returns(postModel);
+
+            var result = await _postService.GetAll(userId);
+
+            result.Should().BeEquivalentTo(expected);
+        }
+
+        [Theory]
+        [MemberData(nameof(GetByIdData))]
+        public async Task Should_get_post(int id, EditPostModel expected)
```

Comment:
nit: the name too similar with the previous one
```suggestion
        public async Task Should_get_specific_post(int id, EditPostModel expected)
```